app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitTrackChairRatingTypesApiController.php

```diff
@@ -0,0 +1,243 @@
+        ISummitRepository $summit_repository,
+        ISelectionPlanRepository $selection_plan_repository,
+        IPresentationTrackChairRatingTypeRepository $repository,
+        ITrackChairRankingService $service,
+        IResourceServerContext $resource_server_context
+    )
+    {
+        parent::__construct($resource_server_context);
+        $this->service = $service;
+        $this->summit_repository = $summit_repository;
+        $this->selection_plan_repository = $selection_plan_repository;
+        $this->repository = $repository;
+    }
+
+    use ParametrizedGetAll;
+
+    /**
+     * @return array
+     */
+    private function getRatingTypeValidationRules(): array
+    {
+        return [
+            'weight'  => 'required|numeric',
+            'name'    => 'required|string',
+            'order'   => 'required|int',
+        ];
+    }
+
+    /**
+     * @return IBaseRepository
+     */
+    protected function getRepository(): IBaseRepository
+    {
+        return $this->repository;
+    }
+
+    /**
+     * @param $summit_id
+     * @param $selection_plan_id
+     * @return \Illuminate\Http\JsonResponse|mixed
+     */
+    public function getTrackChairRatingTypes($summit_id, $selection_plan_id) {
+
+        $summit = SummitFinderStrategyFactory::build($this->summit_repository, $this->resource_server_context)->find($summit_id);
+        if (is_null($summit)) return $this->error404();
+
+        $selection_plan = $this->selection_plan_repository->getById(intval($selection_plan_id));
+        if (is_null($selection_plan)) return $this->error404();
+
+        return $this->_getAll(
+            function () {
+                return [
+                    'selection_plan_id' => ['=='],
+                    'summit_id' => ['=='],
+                ];
+            },
+            function () {
+                return [
+                    'selection_plan_id' => 'sometimes|integer',
+                    'summit_id' => 'sometimes|integer',
+                ];
+            },
+            function () {
+                return [];
+            },
+            function ($filter) use ($summit, $selection_plan) {
+                if ($filter instanceof Filter) {
+                    $filter->addFilterCondition(FilterElement::makeEqual('summit_id', $summit->getId()));
+                    $filter->addFilterCondition(FilterElement::makeEqual('selection_plan_id', $selection_plan->getId()));
+                }
+                return $filter;
+            },
+            function () {
+                $current_user = $this->resource_server_context->getCurrentUser();
+                if(!is_null($current_user)){
+                    if(
+                        $current_user->isOnGroup(IGroup::Administrators) ||
+                        $current_user->isOnGroup(IGroup::SuperAdmins) ||
+                        $current_user->isOnGroup(IGroup::TrackChairsAdmins) ||
+                        $current_user->isOnGroup(IGroup::SummitAdministrators)
```

Comment:
this is not enough u should check the summit->isSummitAdmin

Reply:
done